src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -921,6 +921,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             return;
         }
 
+        // GH#11479: TSF wants to be notified of any character input via ICoreTextEditContext::NotifyTextChanged().
+        // TSF is built and tested around the idea that you inform it of any text changes that happen
+        // when it doesn't currently compose characters. For instance writing "xin chaof" with the
+        // Vietnamese IME should produce "xin chào". After writing "xin" it'll emit a composition
+        // completion event and we'll write "xin" to the shell. It now has no input focus and won't know
+        // about the whitespace. If you then write "chaof", it'll emit another composition completion
+        // event for "xinchaof" and the resulting output in the shell will finally read "xinxinchaof".
+        // A composition completion event technically doesn't mean that the completed text is now

```

Comment:
Wow, this one is correct. IME can change the text in the edit control as it sees fit, you can think of something like autocorrection.
So with the current implementation (before this PR), IME will see a big lump of text without any word delimitation (spaces, dot, comma ...) as they're not notified back to TSF, limiting the ability to provide more advanced text edit functionality that IME can provide.

I have a question regarding the text edit field, is the current command line where the user can input text is not considered a text edit field? And only the composition is considered as a text edit field and has text editing events notified back to TSF?

Reply:
> I have a question regarding the text edit field, is the current command line where the user can input text is not considered a text edit field?

This part is complicated. Right now, we are using an "overlay" text field because we have not connected the command line buffer to the TSF. This is because of how Terminal sees the screen:

|0|1|2|3|4|5|6|
|-|-|-|-|-|-|-|
|C|:|\\ |>|H|i|.|

All of the input _and output_ goes into the same character grid. Terminal does not know that `3` is part of the prompt, and `4` is part of the user input! If we naively sent all that data to TSF, it would look like the user could edit the `C:\>` part.

Reply:
> All of the input _and output_ goes into the same character grid. Terminal does not know that `3` is part of the prompt, and `4` is part of the user input! If we naively sent all that data to TSF, it would look like the user could edit the `C:\>` part.

I see, thank you.
On another hand, I think that we can probably just pass that to the TSF, since the text modification should be depended on the text cursor position as well. And even if the TSF's input processors think it can change the text, if the application doesn't change the text then the input processors should follow.